app/tests/functional/features/steps/pre_signed_url.py

```diff
@@ -0,0 +1,27 @@
+import json
+
+import requests
+from behave import when, then
+from behave.runner import Context
+
+
+@when('I ask for a presigned upload url with this exchange upload token')
+def when_asking_for_pre_signed_upload_url(context: Context):
+    r = requests.get(
+        url='https://dev.api.bda.ninja/s3_presigned_upload_url',
+        headers={
+            'Authorization': context.exchange['upload_token']
+        })
+    context.pre_signed_url_response = r
+    assert 200 == r.status_code, f'HTTP code received is {r.status_code} :('
+
+
+@then("I can upload a file")
+def step_impl(context):
+    pre_signed_url = json.loads(context.pre_signed_url_response.text)
+    form_data = {}
+    for key, value in pre_signed_url['fields'].items():
+        form_data[key] = value
+    form_data['file'] = b'arandombinarycontent'
```

Comment:
Ooh, I got many suggestions for ya:

With a dict comprehension, very fancy:
```suggestion
    form_data = {key: value for key, value in pre_signed_url['fields'].items()}
    form_data['file'] = b'arandombinarycontent'
```

With an unpacked dict, not really a fan tho:
```suggestion
    form_data = dict(**pre_signed_url['fields'], file=b'arandombinarycontent')
```

or my favorite, simpler: 
```suggestion
    form_data = pre_signed_url['fields'].copy()
    form_data['file'] = b'arandombinarycontent'
```